wazimap_ng/datasets/management/commands/reprocess_indicators.py

```diff
@@ -0,0 +1,123 @@
+import sys
+
+from django.core.management.base import BaseCommand, CommandError
+
+from ...models import Indicator
+from django.contrib.auth.models import User
+from django_q.tasks import async_task
+from django_q.brokers import get_broker
+from django_q.models import Task
+
+
+class Command(BaseCommand):
+    help = 'Rerun indicator data process for variables'
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--variables', nargs='+', type=int, default=[],
+            help="pass variable ids to process : --variables 1 2 3"
+        )
+        parser.add_argument(
+            '--datasets', nargs='+', type=int, default=[],
+            help="pass dataset ids to process linked variables : --datasets 1 2 3"
+        )
+        parser.add_argument(
+            '--profiles', nargs='+', type=int, default=[],
+            help="pass profile ids to process linked variables : --profiles 1 2 3"
+        )
+        parser.add_argument(
+            '--user_id', nargs=1, type=int,
+            help="pass user id to get fail task notification : --user_id 2"
+        )
+        parser.add_argument(
+            '--rerun_failed', nargs=1, type=int,
+            help="rerun all failed tasks for extraction-ID group : --rerun_failed 20"
+        )
+
+
+    def handle(self, *args, **options):
+        to_update = options['variables']
+        datasets = options['datasets']
+        profiles = options['profiles']
+        user_id = options['user_id']
+        rerun_failed = options['rerun_failed']
+        email_on_fail = True if user_id else False
+        user = None
+
+        if user_id:
+            user_id = user_id[0]
+            user = User.objects.filter(id=user_id).first()
+            if not user:
+                print(F"User with user id {user_id} does not exist")
+                sys.exit()
+
+        if datasets:
+            to_update = list(Indicator.objects.filter(
+                dataset_id__in=datasets
+            ).values_list("id", flat=True)) + to_update
+
+        if profiles:
+            to_update = list(Indicator.objects.filter(
+                dataset__profile_id__in=profiles
+            ).values_list("id", flat=True)) + to_update
+
+        if rerun_failed:
+            to_update = to_update + [t.args[0].id for t in Task.objects.filter(
+                group=f"extraction-{rerun_failed[0]}", success=False
+            )]
+
+        variables = Indicator.objects.all()
+        if to_update:
+            variables = variables.filter(
+                id__in=list(set(to_update))
+            )
```

Comment:
Why are there dataset and variable filters?

It's not clear what happens when you don't specify these.

We just want a simple way for sysadmins to safely re-process everything. I think you should remove all the fancy stuff.

All this should do is confirm that they are sure they want to reprocess, and then fire away.

Reply:
Done 👍